tests/unit_tests/scheduler/test_scheduler.py

```diff
@@ -468,3 +469,46 @@ async def wait():
         for start, next_start in zip(run_start_times[:-1], run_start_times[1:])
     ]
     assert min(deltas) >= submit_sleep * 0.8
+
+
+@pytest.mark.timeout(5)
+async def test_driver_poll_exceptions_are_propagated(realization):
+    driver = Mock()
+    driver.poll = lambda: 1 / 0
```

Comment:
Ditto wrt. error type. Something like `RuntimeError("Test poll() raises exception")` and match on that entire string further down so that this test doesn't pass if we every _actually_ divide by zero scheduler.